linkup-cli/src/background_booting.rs

```diff
@@ -93,49 +91,66 @@ pub fn load_config(
     Ok(content)
 }
 
-pub fn server_config_from_state(state: &LocalState) -> (StorableSession, StorableSession) {
-    let local_server_services = state
-        .services
-        .iter()
-        .map(|local_service| StorableService {
-            name: local_service.name.clone(),
-            location: if local_service.current == ServiceTarget::Remote {
-                local_service.remote.clone()
-            } else {
-                local_service.local.clone()
-            },
-            rewrites: Some(local_service.rewrites.clone()),
-        })
-        .collect::<Vec<StorableService>>();
-
-    let remote_server_services = state
-        .services
-        .iter()
-        .map(|local_service| StorableService {
-            name: local_service.name.clone(),
-            location: if local_service.current == ServiceTarget::Remote {
-                local_service.remote.clone()
-            } else {
-                state.linkup.tunnel.clone()
-            },
-            rewrites: Some(local_service.rewrites.clone()),
-        })
-        .collect::<Vec<StorableService>>();
-
-    (
-        StorableSession {
+pub struct ServerConfig {
+    pub local: StorableSession,
+    pub remote: StorableSession,
+}
+
+impl From<&LocalState> for ServerConfig {
+    fn from(state: &LocalState) -> Self {
+        let local_server_services = state
+            .services
+            .iter()
+            .map(|local_service| StorableService {
+                name: local_service.name.clone(),
+                location: if local_service.current == ServiceTarget::Remote {
+                    local_service.remote.clone()
+                } else {
+                    local_service.local.clone()
+                },
+                rewrites: Some(local_service.rewrites.clone()),
+            })
+            .collect::<Vec<StorableService>>();
+
+        let remote_server_services = state
+            .services
+            .iter()
+            .map(|local_service| StorableService {
+                name: local_service.name.clone(),
+                location: if local_service.current == ServiceTarget::Remote {
+                    local_service.remote.clone()
+                } else {
+                    state.linkup.tunnel.clone()
+                },
+                rewrites: Some(local_service.rewrites.clone()),
+            })
+            .collect::<Vec<StorableService>>();
```

Comment:
nit: the `local_service` variable name in the map here is confusing, maybe just service?

Reply:
Or switch local to remote in the remote case

Reply:
That's a good point! Changed on 28055f6